controllers/api/userRoutes.js

```diff
@@ -1,61 +1,222 @@
 const router = require('express').Router();
-const { User } = require('../../models');
+const { User, Event, Task } = require('../../models');
+const withAuth = require('../../utils/auth');
 
-router.post('/', async (req, res) => {
-  try {
-    const userData = await User.create(req.body);
+// GET /api/users
+router.get('/', (req, res) => {
+  // Access our User model and run .findAll() method
+  User.findAll({
+      attributes: { exclude: ['email'] }
+  })
+    .then(dbUserData => res.json(dbUserData))
+    .catch(err => {
+      console.log(err);
+      res.status(500).json(err);
+    });
+});
 
-    req.session.save(() => {
-      req.session.user_id = userData.id;
-      req.session.logged_in = true;
+// GET /api/users/1
+router.get('/:id', (req, res) => {
+  User.findOne({
+      attributes: { exclude: ['email']},
+      where: {
+        id: req.params.id
+      },
+      include: [
```

Comment:
The User table does not have a link to the events or tasks that the user is associated with.  Om the pages that we want to display that information will we need to use:
get events where owner_id = req.params.id 
get tasks where volunteer = req.params.id

We probably need to add these two api routes (I don't see them anywhere, except homeRoutes has a '/userProfile' route that tries to pull the event data but I think it will fail).  Probably would want a /user_events route and a /user_tasks route.

Reply:
We can possibly work on this together tomorrow during class.